src/Tes.Runner/Transfer/LinuxSystemInfoProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+
+namespace Tes.Runner.Transfer
+{
+    public class LinuxSystemInfoProvider : ISystemInfoProvider
+    {
+        const string ProcMemInfo = "/proc/meminfo";
+        const string ProcCpuInfo = "/proc/cpuinfo";
+
+        public LinuxSystemInfoProvider()
+        {
+            if (!IsLinuxSystem())
+            {
+                throw new InvalidOperationException("Invalid OS.");
+            }
+        }
+
+        public static bool IsLinuxSystem()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+        }
+
+        public int ProcessorCount
+        {
+            get
+            {
+                return File.ReadLines(ProcCpuInfo).Count(line => line.StartsWith("processor"));
```

Comment:
maybe just do this once, and then store it in a static variable, since this will never change in the process lifetime, and multiple invocations of the file system isn't ideal?